### PR TITLE
SMS

### PR DESCRIPTION
**Pull request by armoj**
Add SMS sending functionality.
* Add new 'confirmation method' screen.
* Add AJV validation for UK mobile phone numbers only
* minor JSON Pointer replacer logic update - allows pointers to potentially not resolve

### CODE REVIEW

File: services/replace-json-pointer/replace-json-pointer.test.js [hunk 1]
```diff
@@ -47,16 +47,26 @@ describe('replace-json-pointer', () => {
...
-        }).toThrow('Invalid reference token: does-not-exist');
+    //     expect(() => {
+    //         replaceJsonPointer(content, {
+    //             foo: {
+    //                 bar: 'bar'
+    //             }
+    //         });
+    //     }).toThrow('Invalid reference token: does-not-exist');
+    // });
+
```
Comment (BarryPiccinni) [inline services/replace-json-pointer/replace-json-pointer.test.js]: Remove if not needed.

File: services/notify/index.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;
```
Comment (BarryPiccinni) [inline services/notify/index.js]: Should this be getting set here??

File: package.json [hunk 3]
```diff
@@ -23,15 +23,19 @@
         "express-jwt": "^5.3.1",
         "express-jwt-authz": "^2.3.1",
         "express-openapi-validator": "^1.6.2",
+        "google-libphonenumber": "^3.2.6",
         "got": "^9.6.0",
-        "helmet": "^3.15.0",
+        "helmet": "^3.21.2",
         "json-pointer": "^0.6.0",
+        "libphonenumber-js": "^1.7.30",
```
Comment (sinclairs) [inline package.json]: Is this and google-libphonenumber required?

File: package.json [hunk 4]
```diff
@@ -75,12 +79,12 @@
...
         "blocks": [
-            ".eslintignore",
             ".eslintrc.js",
-            ".lintstagedrc.js",
+            ".eslintignore",
             ".huskyrc.js",
-            ".prettierrc.js",
             "jest.config.js",
+            ".prettierrc.js",
+            ".lintstagedrc.js",
```
Comment (sinclairs) [inline package.json]: Just wondering if these were manually changed? The base config manages these

File: questionnaire/questionnaire-service.js [hunk 5]
```diff
@@ -10,6 +10,8 @@ const templates = require('./templates');
 const createQuestionnaireDAL = require('./questionnaire-dal');
 const createMessageBusCaller = require('../services/message-bus');
 const replaceJsonPointers = require('../services/replace-json-pointer');
+const createNotifyService = require('../services/notify');
+const createValidatorService = require('../services/validator');
```
Comment (sinclairs) [inline questionnaire/questionnaire-service.js]: Can this be named to communicate what it's validating

File: questionnaire/questionnaire-service.js [hunk 6]
```diff
@@ -23,6 +25,11 @@ function createQuestionnaireService(spec) {
 
     AjvErrors(ajv);
 
+    ajv.addFormat('x-mobilePhoneNumber', mobilePhoneNumber => {
+        const validator = createValidatorService();
```
Comment (sinclairs) [inline questionnaire/questionnaire-service.js]: Do we need a new instance per request?

File: package.json (hunk 3 above)
Comment (armoj) [reply package.json]: removed

File: package.json [hunk 7]
```diff
@@ -23,15 +23,19 @@
...
         "json-pointer": "^0.6.0",
+        "libphonenumber-js": "^1.7.30",
         "lodash.merge": "^4.6.2",
         "morgan": "^1.9.1",
+        "notifications-node-client": "^4.7.0",
         "pg": "^7.11.0",
         "pino": "^5.12.6",
         "pino-http": "^4.2.0",
         "q-router": "github:CriminalInjuriesCompensationAuthority/q-router#v2.4.0",
+        "request": "^2.88.0",
```
[the remark this answers is not in the document]
Comment (armoj) [reply package.json]: `request` used to be a "sub" dependency of `notification-node-client`. This is now not the case. It is needed as a dependency at this level. This is the official recommendation for the `notification-node-client`.

File: package.json (hunk 4 above)
Comment (armoj) [reply package.json]: I don't remember changing these as I know they are managed elsewhere. I've changed them back to the order they were originally.

File: questionnaire/questionnaire-service.js (hunk 6 above)
Comment (armoj) [reply questionnaire/questionnaire-service.js]: `ajv` is declared and initialised just above this within the same scope. This is the earliest the new format can be defined. Are you suggesting a bigger change here than moving this `addFormat` block?

File: questionnaire/questionnaire-service.js [hunk 8]
```diff
@@ -23,6 +25,11 @@ function createQuestionnaireService(spec) {
 
     AjvErrors(ajv);
 
+    ajv.addFormat('x-mobilePhoneNumber', mobilePhoneNumber => {
+        const mobilePhoneNumberValidator = createMobilePhoneNumberValidatorService();
+        return mobilePhoneNumberValidator.isValidMobilePhoneNumber(mobilePhoneNumber);
+    });
+
```
Comment (armoj) [inline questionnaire/questionnaire-service.js]: @sinclairs `ajv` is defined just above this within the same scope. This is the earliest that I can define a new "format" for AJV. Is this what you were originally meaning or is there another issue you were talking about?

**Pull request merged**